src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs

```diff
@@ -54,30 +54,67 @@ public FamilyNameCollection(string familyNames)
         /// </value>
         internal IReadOnlyList<string> Names { get; }
 
-        /// <inheritdoc />
         /// <summary>
-        /// Returns an enumerator that iterates through the collection.
+        /// Returns an enumerator for the name collection.
         /// </summary>
-        /// <returns>
-        /// An enumerator that can be used to iterate through the collection.
-        /// </returns>
-        public IEnumerator<string> GetEnumerator()
+        public Enumerator GetEnumerator()
         {
-            return Names.GetEnumerator();
+            return new Enumerator(this);
+        }
+
+        IEnumerator<string> IEnumerable<string>.GetEnumerator()
+        {
+            return GetEnumerator();
         }
 
-        /// <inheritdoc />
-        /// <summary>
-        /// Returns an enumerator that iterates through a collection.
-        /// </summary>
-        /// <returns>
-        /// An <see cref="T:System.Collections.IEnumerator"></see> object that can be used to iterate through the collection.
-        /// </returns>
         IEnumerator IEnumerable.GetEnumerator()
         {
             return GetEnumerator();
         }
 
+        public struct Enumerator : IEnumerator, IEnumerator<string>
```

Comment:
It might be better to extract this enumerator to some utility namespace. `ReadOnlyListStuctEnumerator<T>` or something.

Reply:
@kekekeks I did this change. Please let me know if that's what you wanted.